packages/connect-agreement/src/entities/Agreement.ts

```diff
@@ -0,0 +1,74 @@
+import { Address, SubscriptionHandler } from '@aragon/connect-types'
+
+import Signer from './Signer'
+import Version from './Version'
+import { IAgreementConnector } from '../types'
+
+export default class Agreement {
+  #appAddress: Address
+  #connector: IAgreementConnector
+
+  constructor(connector: IAgreementConnector, appAddress: Address) {
+    this.#connector = connector
+    this.#appAddress = appAddress
+  }
+
+  async disconnect() {
+    await this.#connector.disconnect()
+  }
+
+  async id(): Promise<string> {
+    const data = await this.#connector.agreement(this.#appAddress)
+    return data.id
+  }
```

Comment:
Just a note on not having `onX` equivalents for some of the methods: I think it makes sense, some methods don’t need to receive updates so it would be a bit odd to ask connector authors to implement them.

It would still be useful to always be able to subscribe (so we could add e.g. a retry mechanism), and to provide a unified API, but this is something we can do at a higher level, e.g. in `connect-core`.